File: programs/advanced-topics_more-constructors-4.cpp
RegularPolygon(int nsides, int radius, string color) {
  // validate the arguments
  if (nsides < 3) {
    throw invalid_argument("invalid nsides");
  }
  if (radius <= 0) {
    throw invalid_argument("invalid radius");
  }
  if (color.empty()) {
    throw invalid_argument("invalid color");
  }

  // the rest of the method is omitted
}

File: programs/advanced-topics_more-constructors-4.java
public RegularPolygon(int nsides, int radius, Color color) {

    // validate the arguments
    if (nsides &lt; 3) {
        throw new IllegalArgumentException("invalid nsides");
    }
    if (radius &lt;= 0) {
        throw new IllegalArgumentException("invalid radius");
    }
    if (color == null) {
        throw new NullPointerException("invalid color");
    }

    // the rest of the method is omitted